presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/environment/SinglenodeKafka.java

```diff
@@ -0,0 +1,87 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.prestosql.tests.product.launcher.env.environment;
+
+import com.google.common.collect.ImmutableList;
+import io.prestosql.tests.product.launcher.docker.DockerFiles;
+import io.prestosql.tests.product.launcher.env.DockerContainer;
+import io.prestosql.tests.product.launcher.env.Environment;
+import io.prestosql.tests.product.launcher.env.common.AbstractEnvironmentProvider;
+import io.prestosql.tests.product.launcher.env.common.Standard;
+
+import javax.inject.Inject;
+
+import static io.prestosql.tests.product.launcher.env.common.Hadoop.CONTAINER_PRESTO_CATALOG;
+import static io.prestosql.tests.product.launcher.env.common.Hadoop.CONTAINER_PRESTO_KAFKA_PROPERTIES;
+import static java.util.Objects.requireNonNull;
+import static org.testcontainers.containers.BindMode.READ_ONLY;
+
+public class SinglenodeKafka
+        extends AbstractEnvironmentProvider
+{
+    private static final String CONFLUENT_VERSION = "5.2.1";
+
+    private final DockerFiles dockerFiles;
+
+    @Inject
+    public SinglenodeKafka(Standard standard, DockerFiles dockerFiles)
+    {
+        super(ImmutableList.of(standard));
+        this.dockerFiles = requireNonNull(dockerFiles, "dockerFiles is null");
+    }
+
+    @Override
+    protected void extendEnvironment(Environment.Builder builder)
+    {
+        super.extendEnvironment(builder);
+
+        builder.configureContainer("presto-master", container -> container
+                .withFileSystemBind(dockerFiles.getDockerFilesHostPath("conf/environment/singlenode-kafka/kafka.properties"), CONTAINER_PRESTO_KAFKA_PROPERTIES, READ_ONLY))
+                .withFileSystemBind(dockerFiles.getDockerFilesHostPath("conf/environment/singlenode-kafka/kafka"), CONTAINER_PRESTO_CATALOG + "kafka", READ_ONLY));
+
+        builder.configureContainer("tests", container -> container
+                .withFileSystemBind(dockerFiles.getDockerFilesHostPath("conf/environment/singlenode-kafka/kafka"), CONTAINER_PRESTO_CATALOG + "kafka", READ_ONLY));
+
+        builder.addContainer("zookeeper", createZookeeper());
+        builder.addContainer("kafka", createKafka());
+    }
+
+    @SuppressWarnings("resource")
+    private DockerContainer createZookeeper()
+    {
+        DockerContainer container = new DockerContainer("confluentinc/cp-zookeeper:" + CONFLUENT_VERSION)
+                .withNetworkAliases("zookeeper")
+                .withEnv("ZOOKEEPER_CLIENT_PORT", "2181")
+                .withEnv("ZOOKEEPER_TICK_TIME", "2000")
+                .withExposedPorts(2181);
```

Comment:
the port gets exposed on localhost at random number

that's unlikely useful. Skip it?
Or do you use it as an implicit health check? (yes =>  add a comment)